pytorch_lightning/utilities/distributed.py

```diff
@@ -155,3 +159,61 @@ def sync_ddp(
         result = result / torch.distributed.get_world_size(group)
 
     return result
+
+
+class AllGatherGrad(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, tensor, group=group.WORLD):
+        ctx.batch_size = tensor.shape[0]
+        ctx.group = group
+
+        gathered_tensor = [
```

Comment:
Could you extend this to support tuple of tensors ?

Reply:
tuple of tensors how? Like gather a tuple of tensors across processes?

Reply:
@tchaton I think we shouldn't do this. IMO This should only be working on a single tensor and if we want to support tuples, we should just call this function onto the tuples' elements (e.g. with `apply_to_collection`), but we really shouldn't include this in that function here!